src/main/java/com/mainstreethub/ecr/LoginMojo.java

```diff
@@ -0,0 +1,62 @@
+package com.mainstreethub.ecr;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.ecr.AmazonECR;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import com.amazonaws.services.ecr.model.AuthorizationData;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenRequest;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenResult;
+import com.google.common.base.Splitter;
+import com.google.common.io.BaseEncoding;
+import java.util.List;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+@Mojo(name = "login")
+public class LoginMojo extends AbstractMojo {
+  private static final Splitter TOKEN_SPLITTER = Splitter.on(":").limit(3).trimResults();
```

Comment:
For a splitter I usually limit to expected size + 1. This is so if I get
more of the separator character than expected I will know.

So, a string like this will trigger the failure condition:

user:password:something

Granted, it will also fail when I do the base64 encoding as : isn't a
valued base64 character, but this way has usually been clearer in my mind
at least. :)

Thoughts? Should I stick with 2 and just let the decoder fail if need be?
On Apr 21, 2016 9:08 PM, "Brandon Beck" <notifications@github.com> wrote:

> In src/main/java/com/mainstreethub/ecr/LoginMojo.java
> <https://github.com/mainstreethub/maven-ecr-plugin/pull/1#discussion_r60683010>
> :
>
> > +import com.amazonaws.services.ecr.model.AuthorizationData;
> > +import com.amazonaws.services.ecr.model.GetAuthorizationTokenRequest;
> > +import com.amazonaws.services.ecr.model.GetAuthorizationTokenResult;
> > +import com.google.common.base.Splitter;
> > +import com.google.common.io.BaseEncoding;
> > +import java.util.List;
> > +import org.apache.maven.plugin.AbstractMojo;
> > +import org.apache.maven.plugin.MojoExecutionException;
> > +import org.apache.maven.plugin.MojoFailureException;
> > +import org.apache.maven.plugins.annotations.Mojo;
> > +import org.apache.maven.plugins.annotations.Parameter;
> > +import org.apache.maven.project.MavenProject;
> > +
> > +@Mojo(name = "login")
> > +public class LoginMojo extends AbstractMojo {
> > +  private static final Splitter TOKEN_SPLITTER = Splitter.on(":").limit(3).trimResults();
>
> Shouldn't this be a limit of 2 based on the code below expecting two
> elements?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mainstreethub/maven-ecr-plugin/pull/1/files/43f08beb61797e872555cdc232e02d7439b0415c#r60683010>
>